tree/treeplayer/src/TSimpleAnalysis.cxx

```diff
@@ -0,0 +1,252 @@
+      ptrHisto->Write();
+      if (ptrHisto == nullptr)
+         return false;
+   }
+   return true;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Return false if not a tree name
+///
+/// param[in] line line read from the input file
+
+static bool HandleTreeNameConfig(const std::string& line)
+{
+   return line.find("=") == std::string::npos;
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// Skip subsequent empty lines in a string and returns the number of the
+/// current line of the input file
+///
+/// param[in] numbLine number of the input file line
+
+std::string TSimpleAnalysis::SkipSubsequentEmptyLines(int& numbLine)
+{
+   std::string notEmptyLine;
+
+   while (getline(fIn,notEmptyLine) && DeleteSpaces(notEmptyLine)
+          && (notEmptyLine.empty() || notEmptyLine.find("#") == 0 ||
+              notEmptyLine.find_first_not_of(" ") == std::string::npos))
+      {numbLine++;}
+   numbLine++;
+   return notEmptyLine;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// After the eventual skip of subsequent empty lines, returns true if the
+/// line is a comment
+///
+/// param[in] line line read from the input file
+/// param[in] readingSection current section of the read file
+/// param[in] numbLine number of the input file line
+
+bool TSimpleAnalysis::HandleLines(std::string& line, int& readingSection, int& numbLine)
+{
+   if (line.empty() || line.find_first_not_of(" ") == std::string::npos) {
+      if (readingSection == 0 && fCounter == 0)
+         return 1;
+      readingSection++;
+      line = SkipSubsequentEmptyLines(numbLine);
+   }
+
+   DeleteSpaces(line);
+   std::size_t comment = line.find("#");
+   if (comment == 0)
+      return 1;
+   if (((comment != 0) || (comment != std::string::npos)) && readingSection == 0)
+      fCounter++;
+   line = line.substr(0, comment);
+   DeleteSpaces(line);
+   return 0;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// This function has the aim of setting the arguments read from the input file
+
+bool TSimpleAnalysis::Configure()
+{
+   int readingSection = kReadingOutput;
+   std::string line;
+   int numbLine = 0;
+
+   fIn.open(fInputName);
+   if(!fIn) {
+      ::Error("TSimpleAnalysis", "File %s not found", fInputName.c_str());
+      return 0;
+   }
+
+   while(!fIn.eof()) {
+      std::string errMessage;
+
+      getline (fIn,line);
+      numbLine++;
+      if (HandleLines(line, readingSection, numbLine))
+         continue;
+
+         switch (readingSection) {
+         case kReadingOutput:
+            fOutputFile = line;
+            break;
+
+         case kReadingInput:
+            fInputFiles.push_back(line);
+            break;
+
+         case kReadingTreeName:
+            if (HandleTreeNameConfig(line)) {
+               fTreeName = line;
```

Comment:
Move this into `HandleTreeNameConfig()` - it claims to *handle* the config.
Also: check whether name is empty and if so, error!